example_plugins/foo/src/index.ts

```diff
@@ -1,16 +1,27 @@
 import { KibanaPluginConfig } from '@elastic/kbn-types';
+import { k$, first, toPromise } from '@elastic/kbn-observable';
 import { BarPluginType } from 'example-plugin-bar';
+import { FooConfig } from './FooConfig';
 
 export const plugin: KibanaPluginConfig<BarPluginType> = {
-  configPath: ['foo'],
+  configPath: ['__newPlatform', 'foo'],
```

Comment:
__note:__ I tried two approaches here:

1) Hide this `__newPlatform` prefix inside our plugin management infrastructure and always prepend plugin's config path with `__newPlatform`;
2) Allow every plugin to decide whether it wants to store its config under unstable `__newPlatform` or not.

I chose 2nd approach as it seems to be the most flexible: once plugin is ready to replace its counterpart from the legacy platform, it just removes `__newPlatfrom` from its `configPath` and defines its config schema in the legacy config, other non-stable-yet plugins won't affected.